pkg/apis/sources/v1alpha1/awss3_types.go

```diff
@@ -65,19 +65,29 @@ type AWSS3SourceSpec struct {
 	// https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-event-types-and-destinations.html
 	EventTypes []string `json:"eventTypes"`
 
-	// SQS Queue ARN
-	// https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonsqs.html#amazonsqs-resources-for-iam-policies
-	//
-	// When not provided, a SQS queue for receiving event notifications
-	// from the S3 bucket is automatically created and configured.
+	// The destination of notifications originating from the Amazon S3 bucket.
 	//
-	// +optional
-	QueueARN *apis.ARN `json:"queueARN,omitempty"`
+	// If omitted, a SQS queue is automatically created and associated with
+	// the bucket.
+	Destination *AWSS3SourceDestination `json:"destination,omitempty"`
```

Comment:
Shouldn't the `// +optional` annotation be included here as well?

Reply:
Nice catch!